main.tf

```diff
@@ -175,3 +177,25 @@ resource "gitlab_branch_protection" "default" {
     }
   }
 }
+
+################################################################################
+# Pipeline schedule
+################################################################################
+resource "gitlab_pipeline_schedule" "default" {
+  count  = var.pipeline_schedule.cron != null ? 1 : 0
```

Comment:
```
locals {
  # On/Off switches for optional resources and configuration
  pipeline_schedule_enabled = var.pipeline_schedule != null ? true : false
}

resource "gitlab_pipeline_schedule" "default" {
  for_each = local.pipeline_schedule_enabled ? { instance = true } : {}

  active         = var.pipeline_schedule.active
  cron           = var.pipeline_schedule.cron
  cron_timezone  = var.pipeline_schedule.cron_timezone
  description    = var.pipeline_schedule.description
  project        = gitlab_project.default.id
  ref            = var.pipeline_schedule.ref
  take_ownership = var.pipeline_schedule.take_ownership

  lifecycle {
    precondition {
      condition     = can(regex("^([0-5]?[0-9]|\\*) ([0-9]|1[0-9]|2[0-3]|\\*) ([1-9]|[12][0-9]|3[01]|\\*) ([1-9]|1[0-2]|\\*) ([0-6]|\\*)$", var.pipeline_schedule.cron))
      error_message = "The cron expression is not valid. It should be in the format '0 1 * * *'."
    }
  }
}
```

Reply:
modified the "if condition" a bit this will not work anymore due to the non optional description and we try to move away from count in our mcaf modules and use for_each consistently so making that change as well.